File: kthsmallest.cpp
Time Complexity - O(nklogk) where n is the total number of elements in the matrix
Space Complexity - O(k)
Ran successfully on leetcode

class Solution {
public:
    int kthSmallest(vector<vector<int>>& matrix, int k) {
        //base
        if(matrix.empty())
            return -1;
        priority_queue<int>pq;
        for(int i=0;i<matrix.size();i++)
            for(int j=0;j<matrix[0].size();j++)
            {
                pq.push(matrix[i][j]);
                if(pq.size()>k)
                    pq.pop();
            }
        return pq.top();
    }
};

File: kthSmallest.py

'''
time complexity: o(nlogn)
n is total elements
space complexity O(n) //space occupied by heap
'''
import heapq
class Solution:
    def kthSmallest(self, matrix: List[List[int]], k: int) -> int:
        
        ls = []
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                heapq.heappush(ls,-matrix[i][j])
                if(len(ls)>k):
                    heapq.heappop(ls)
        return -ls[0]

        